eng/common/pipelines/templates/steps/check-spelling.yml

```diff
@@ -26,3 +38,21 @@ steps:
           -CspellConfigPath ${{ parameters.CspellConfigPath }}
           -ExitWithError:(!$${{ parameters.ContinueOnError }})
         pwsh: true
+
+    - pwsh: |
+        $changedFiles = ./eng/common/scripts/get-changedfiles.ps1
+
+        if ($changedFiles -notcontains 'eng/common/spelling/package-lock.json') {
+          Write-Host "No changes to cspell package-lock.json detected."
+          exit 0
+        }
+
+        Write-Host "Detected change to cspell package-lock.json. Running upgrade verification."
+        & '${{ parameters.ScriptToValidateUpgrade }}'
+      displayName: Verify cspell upgrade
+      condition: >-
+        and(
+          succeeded(), 
+          ne('true', variables['Skip.SpellCheck']),
+          ne('', parameters.ScriptToValidateUpgrade)
```

Comment:
nit: Might be better to have this condition to be a template condition so this step doesn't show up for any repos other than js/.NET.